### PR TITLE
[SPARK-40799] [BUILD] [CONNECT] Enforce scalafmt for Spark Connect module.

### PR DESCRIPTION
**Pull request by grundprinzip**
### What changes were proposed in this pull request?
This changes enables enforcing `scalafmt` for the Spark Connect module to have a consistent enforced coding style and reduce unnecessary code-review cycles. This patch specifically only enforces this on the Spark Connect module to avoid impact on the other areas of the Spark codebase.

To pass the pre-commit, this change applies `scalafmt` on the existing code-base.

### Why are the changes needed?
Faster, focussed code reviews.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Manual testing / CI

### CODE REVIEW

**Review by hvanhovell — APPROVED** [review]
LGTM

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Will merge this in a few days.

**Comment by amaliujia** [comment]
Will the github precommit fail if scala formatter is not applied yet in a PR?

**Comment by grundprinzip** [comment]
> Will the github precommit fail if scala formatter is not applied yet in a PR?

Yes, this integrates in the check that is run as part of the dev/lint-scala script.

**Comment by amaliujia** [comment]
> > Will the github precommit fail if scala formatter is not applied yet in a PR?
> 
> Yes, this integrates in the check that is run as part of the dev/lint-scala script.

Nice! LGTM!

File: dev/lint-scala [hunk 1]
```diff
@@ -21,3 +21,13 @@ SCRIPT_DIR="$( cd "$( dirname "$0" )" && pwd )"
 SPARK_ROOT_DIR="$(dirname $SCRIPT_DIR)"
 
 "$SCRIPT_DIR/scalastyle" "$1"
+
+# For Spark Connect, we actively enforce scalafmt and check that the produced diff is empty.
+./build/mvn -Pscala-2.12 scalafmt:format -Dscalafmt.skip=false -Dscalafmt.validateOnly=true -pl connector/connect
+if [[ $? -ne 0 ]]; then
+  echo "The scalafmt check failed on connector/connect."
+  echo "Before submitting your change, please make sure to format your code using the following command:"
+  echo "./build/mvn -Pscala-2.12 scalafmt:format -Dscalafmt.skip=fase -Dscalafmt.validateOnly=false -pl connector/connect"
```
Comment (amaliujia) [inline dev/lint-scala:30]: :) No one loves SBT

Comment (amaliujia) [reply dev/lint-scala:30]: I am not familiar with SBT though maybe this works with SBT.

Comment (grundprinzip) [reply dev/lint-scala:30]: This piece here is taken from the existing `dev/scalafmt` script and slightly adjusted to only reformat the Spark Connect module. I don't think we've ported the usage of `scalafmt` to SBT.

**Comment by grundprinzip** [comment]
@HyukjinKwon let me know when you're ready to merge and I will address the merge conflicts by running `scalafmt` again to fix the existing issues.

**Comment by grundprinzip** [comment]
Checking that the build will fail because the lint fails

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by HyukjinKwon** [comment]
ah, I am sorry it was my bad. I didn't see the last changes made. Let me revert and reopen.

**Comment by grundprinzip** [comment]
Verified that the pre-commit is failing appropriately.

**Comment by HyukjinKwon** [comment]
Merged to master.

File: dev/lint-scala [hunk 2]
```diff
@@ -21,3 +21,12 @@ SCRIPT_DIR="$( cd "$( dirname "$0" )" && pwd )"
...
 "$SCRIPT_DIR/scalastyle" "$1"
+
+# For Spark Connect, we actively enforce scalafmt and check that the produced diff is empty.
+./build/mvn -Pscala-2.12 scalafmt:format -Dscalafmt.skip=false -Dscalafmt.validateOnly=true -Dscalafmt.changedOnly=false -pl connector/connect
+if [[ $? -ne 0 ]]; then
+  echo "The scalafmt check failed on connector/connect."
+  echo "Before submitting your change, please make sure to format your code using the following command:"
+  echo "./build/mvn -Pscala-2.12 scalafmt:format -Dscalafmt.skip=fase -Dscalafmt.validateOnly=false -Dscalafmt.changedOnly=false -pl connector/connect"
+  exit 1
+fi
```
Comment (dongjoon-hyun) [inline dev/lint-scala:32]: Hi, @grundprinzip and @HyukjinKwon .

This PR makes `lint-scala` ignore the exit code of `scalastyle`.

**Comment by dongjoon-hyun** [comment]
I made a fix here.
- #49443

File: dev/lint-scala (hunk 2 above)
Comment (grundprinzip) [reply dev/lint-scala:32]:
```
if [[ $? -ne 0 ]]; then
```
This checks explicitly for a non-zero exit code. Why would this not cover your case?

**Pull request closed without merging**